packages/tobot_engine/src/core/exit_codes.cpp

```diff
@@ -0,0 +1,14 @@
+#include "exit_codes.h"
+
+using namespace Tobot::Core;
+
+int ExitCode::getCode() const {
+    return this->code;
+}
+char const * ExitCode::getDescription() const {
+    return this->description;
+}
+const ExitCode ExitCode::OK = ExitCode(0, "Everything went well");
```

Comment:
Maybe you can check out this here. https://github.com/felangel/mason/blob/master/packages/mason_logger/lib/src/io.dart

This is what i use for my exitcode when building a cli in dart

Reply:
Looks like this implementation is based on the exit codes under unix as well.
## Sysexits.h under ubuntu
#define EX_OK           0       /* successful termination */

#define EX__BASE        64      /* base value for error messages */

#define EX_USAGE        64      /* command line usage error */
#define EX_DATAERR      65      /* data format error */
#define EX_NOINPUT      66      /* cannot open input */
#define EX_NOUSER       67      /* addressee unknown */
#define EX_NOHOST       68      /* host name unknown */
#define EX_UNAVAILABLE  69      /* service unavailable */
#define EX_SOFTWARE     70      /* internal software error */
#define EX_OSERR        71      /* system error (e.g., can't fork) */
#define EX_OSFILE       72      /* critical OS file missing */
#define EX_CANTCREAT    73      /* can't create (user) output file */
#define EX_IOERR        74      /* input/output error */
#define EX_TEMPFAIL     75      /* temp failure; user is invited to retry */
#define EX_PROTOCOL     76      /* remote error in protocol */
#define EX_NOPERM       77      /* permission denied */
#define EX_CONFIG       78      /* configuration error */

#define EX__MAX 78      /* maximum listed value */

Reply:
Alright, superb. Could you please also add the rest for (possible) future use? 🙂

Reply:
sure